server/rooms.ts

```diff
@@ -1054,103 +1037,85 @@ class BasicChatRoom extends BasicRoom {
 		}
 		this.log = Roomlogs.create(this, options);
 
-		/** @type {any} */
-		// TODO: strongly type polls
 		this.poll = null;
 
 		// room settings
 		this.desc = '';
-		this.modchat = (Config.chatmodchat || false);
+		this.modchat = (Config.chatmodchat || null);
 		this.filterStretching = false;
 		this.filterEmojis = false;
 		this.filterCaps = false;
-		/** @type {false | number} */
 		this.slowchat = false;
 		this.introMessage = '';
 		this.staffMessage = '';
 		this.autojoin = false;
-		this.staffAutojoin = /** @type {string | boolean} */ (false);
+		this.staffAutojoin = false;
 
-		/** @type {?true | RegExp} */
 		this.banwordRegex = null;
-		/** @type {string[]} */
 		this.banwords = [];
 
-		/** @type {number?} */
 		// Only available in groupchats
 		this.uptime = null;
 
 		this.chatRoomData = (options.isPersonal ? null : options);
 		Object.assign(this, options);
 		if (this.auth) Object.setPrototypeOf(this.auth, null);
-		/** @type {Room?} */
 		this.parent = null;
 		if (options.parentid) {
 			const parent = Rooms(options.parentid);
 
 			if (parent) {
 				if (!parent.subRooms) parent.subRooms = new Map();
-				parent.subRooms.set(this.id, /** @type {ChatRoom} */ (this));
+				parent.subRooms.set(this.id, this as ChatRoom);
 				this.parent = parent;
 			}
 		}
 
-		/** @type {Map<string, ChatRoom>?} */
 		this.subRooms = null;
 
-		/** @type {'chat' | 'battle'} */
 		this.type = 'chat';
-		/** @type {boolean} */
 		this.active = false;
 		// TypeScript bug: subclass null
```

Comment:
```suggestion
```

This isn't a problem in real TypeScript, as you can tell by how you could remove the doc comment.